### PR TITLE
Improved the output in minikube github action

### LINKED ISSUE #13233: Progress bar in Github Action polutes the output
### What Happened?

when running minikube in github action I noticed in Ubuntu it polutes the output which is not pretty
but in github action on macOS it is doing the right thing

example can be found here

https://github.com/medyagh/test-minikube-example/runs/4613732249?check_suite_focus=true

### Attach the log file

on linux 

```
Run medyagh/setup-minikube@latest
/usr/bin/chmod +x /home/runner/work/_temp/dd71f544-68b8-49d8-91bb-9d85c09f0f74
/home/runner/bin/minikube start --wait all
* minikube v1.24.0 on Ubuntu 20.04
* Automatically selected the docker driver. Other choices: podman, none, ssh
* Starting control plane node minikube in cluster minikube
* Pulling base image ...
* Downloading Kubernetes v1.22.3 preload ...
    > gcr.io/k8s-minikube/kicbase: 24.36 KiB / 355.78 MiB [>____] 0.01% ? p/s ?    > gcr.io/k8s-minikube/kicbase: 18.84 MiB / 355.78 MiB [>____] 5.29% ? p/s ?    > gcr.io/k8s-minikube/kicbase: 27.27 MiB / 355.78 MiB [>____] 7.67% ? p/s ?    > gcr.io/k8s-minikube/kicbase: 27.28 MiB / 355.78 MiB  7.67% 45.37 MiB p/s     > gcr.io/k8s-minikube/kicbase: 42.74 MiB / 355.78 MiB  12.01% 45.37 MiB p/s    > gcr.io/k8s-minikube/kicbase: 48.76 MiB / 355.78 MiB  13.70% 45.37 MiB p/s    > gcr.io/k8s-minikube/kicbase: 67.83 MiB / 355.78 MiB  19.06% 46.80 MiB p/s    > gcr.io/k8s-minikube/kicbase: 84.48 MiB / 355.78 MiB  23.75% 46.80 MiB p/s    > gcr.io/k8s-minikube/kicbase: 97.53 MiB / 355.78 MiB  27.41% 46.80 MiB p/s    > gcr.io/k8s-minikube/kicbase: 132.31 MiB / 355.78 MiB  37.19% 50.72 MiB p/    > gcr.io/k8s-minikube/kicbase: 167.03 MiB / 355.78 MiB  46.95% 50.72 MiB p/    > gcr.io/k8s-minikube/kicbase: 203.80 MiB / 355.78 MiB  57.28% 50.72 MiB p/    > gcr.io/k8s-minikube/kicbase: 208.11 MiB / 355.78 MiB  58.49% 55.60 MiB p/    > gcr.io/k8s-minikube/kicbase: 208.11 MiB / 355.78 MiB  58.49% 55.60 MiB p/    > gcr.io/k8s-minikube/kicbase: 216.11 MiB / 355.78 MiB  60.74% 55.60 MiB p/    > gcr.io/k8s-minikube/kicbase: 247.16 MiB / 355.78 MiB  69.47% 56.19 MiB p/    > gcr.io/k8s-minikube/kicbase: 256.89 MiB / 355.78 MiB  72.20% 56.19 MiB p/    > gcr.io/k8s-minikube/kicbase: 288.64 MiB / 355.78 MiB  81.13% 56.19 MiB p/    > gcr.io/k8s-minikube/kicbase: 302.60 MiB / 355.78 MiB  85.05% 58.54 MiB p/    > gcr.io/k8s-minikube/kicbase: 330.05 MiB / 355.78 MiB  92.77% 58.54 MiB p/    > gcr.io/k8s-minikube/kicbase: 347.49 MiB / 355.78 MiB  97.67% 58.54 MiB p/    > gcr.io/k8s-minikube/kicbase: 347.49 MiB / 355.78 MiB  97.67% 59.60 MiB p/    > gcr.io/k8s-minikube/kicbase: 347.49 MiB / 355.78 MiB  97.67% 59.60 MiB p/    > gcr.io/k8s-minikube/kicbase: 347.49 MiB / 355.78 MiB  97.67% 59.60 MiB p/    > gcr.io/k8s-minikube/kicbase: 355.74 MiB / 355.78 MiB  99.99% 56.64 MiB p/    > gcr.io/k8s-minikube/kicbase: 355.75 MiB / 355.78 MiB  99.99% 56.64 MiB p/    > gcr.io/k8s-minikube/kicbase: 355.75 MiB / 355.78 MiB  99.99% 56.64 MiB p/    > gcr.io/k8s-minikube/kicbase: 355.75 MiB / 355.78 MiB  99.99% 52.99 MiB p/    > gcr.io/k8s-minikube/kicbase: 355.76 MiB / 355.78 MiB  99.99% 52.99 MiB p/    > gcr.io/k8s-minikube/kicbase: 355.77 MiB / 355.78 MiB  100.00% 52.99 MiB p    > gcr.io/k8s-minikube/kicbase: 355.77 MiB / 355.78 MiB  100.00% 49.57 MiB p    > gcr.io/k8s-minikube/kicbase: 355.77 MiB / 355.78 MiB  100.00% 57.93 MiB p* Creating docker container (CPUs=2, Memory=2200MB) ...
* Preparing Kubernetes v1.22.3 on Docker 20.10.8 ...
  - Generating certificates and keys ...
  - Booting up control plane ...
  - Configuring RBAC rules ...
* Verifying Kubernetes components...
  - Using image gcr.io/k8s-minikube/storage-provisioner:v5
* Enabled addons: default-storageclass, storage-provisioner
* Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
```

but on macos it is pretty

```
Run medyagh/setup-minikube@latest
/bin/chmod +x /Users/runner/work/_temp/d2790973-7b03-436e-826b-ecd2ebc57e21
/Users/runner/bin/minikube start --wait all
* minikube v1.24.0 on Darwin 10.15.7
* Automatically selected the virtualbox driver
* Downloading VM boot image ...
* Starting control plane node minikube in cluster minikube
* Downloading Kubernetes v1.22.3 preload ...
* Creating virtualbox VM (CPUs=2, Memory=3500MB, Disk=20000MB) ...
* Preparing Kubernetes v1.22.3 on Docker 20.10.8 ...
  - Generating certificates and keys ...
  - Booting up control plane ...
  - Configuring RBAC rules ...
* Verifying Kubernetes components...
  - Using image gcr.io/k8s-minikube/storage-provisioner:v5
* Enabled addons: storage-provisioner, default-storageclass
* Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
```

### Operating System

_No response_

### Driver

_No response_

### PR DESCRIPTION
**Pull request by NikhilSharmaWe**
Fixes #13233

### CODE REVIEW

File: pkg/minikube/download/download.go [hunk 1]
```diff
@@ -60,7 +61,7 @@ func CreateDstDownloadMock(src, dst string) error {
 // download is a well-configured atomic download function
 func download(src string, dst string) error {
 	var clientOptions []getter.ClientOption
-	if out.IsTerminal(os.Stdout) {
+	if out.IsTerminal(os.Stdout) && !(detect.GithubActionRunner()) {
```
Comment (NikhilSharmaWe) [inline pkg/minikube/download/download.go:64]: @spowelljr added the condition for using progress bar when terminal is not in Github Actions.

File: pkg/minikube/detect/detect.go [hunk 2]
```diff
@@ -107,3 +107,9 @@ func DockerInstalledViaSnap() bool {
 
 	return strings.Contains(string(o), "snap")
 }
+
+// GithubActionRunner returns true if running inside a github action runner
+func GithubActionRunner() bool {
```
Comment (spowelljr) [inline pkg/minikube/detect/detect.go:112]: This func should be removed from the test file you copied it from and the tests that reference it should be updated to use this one.

**Review by spowelljr — CHANGES_REQUESTED** [review]

**Comment by NikhilSharmaWe** [comment]
@spowelljr Could you please review the changes.

File: pkg/minikube/download/download.go (hunk 1 above)
Comment (spowelljr) [inline pkg/minikube/download/download.go:64]:
```suggestion
	if out.IsTerminal(os.Stdout) && !detect.GithubActionRunner() {
```
The logic is fine, but you don't have to have `()` around the func, so lets remove those to stay consistent with the rest of the code base and after that looks good to merge to me!

**Review by spowelljr — CHANGES_REQUESTED** [review]

**Review by spowelljr — APPROVED** [review]
Looks good to me, thanks the PR!

**Pull request merged**